src/engine/Service.cpp

```diff
@@ -503,8 +505,9 @@ void Service::precomputeSiblingResult(std::shared_ptr<Operation> left,
   AD_CORRECTNESS_CHECK(left && right);
 
   auto skipSortOperation = [](std::shared_ptr<Operation>& op) {
-    auto children = op->getChildren();
-    if (children.size() == 1) {
+    if (static_cast<bool>(std::dynamic_pointer_cast<Sort>(op))) {
+      auto children = op->getChildren();
```

Comment:
```suggestion
      const auto& children = op->getChildren();
```
probably suffices (assuming that `getRootOperation` is const. else leave it and tell me).